### ISSUE TITLE
`dagster/priority` in asset's `op_tags` has no effect

### ISSUE THREAD

**Issue by mmutso-boku** [dagster-io/dagster#25129]
### What's the issue?

Basically, the behaviour/feature describe in https://docs.dagster.io/guides/limiting-concurrency-in-data-pipelines#setting-priority-for-opsassets does not work. More specifically the example `Priority can also be assigned based on the priority tag of the op/asset definition.`

When defining the priority op_tag for an asset, this seems to have no effect in any scenario - priority tag on the run vs on the asset op_tags; two assets's priority op_tag compared to each other etc.

The following statement in the docs also seems to be incorrect: `When the priority tag is set on both the run and the asset/op definition, the step priority will be the sum of the two values.` 

### What did you expect to happen?

I expected the `dagster/priority` tag to reach the run in one way or another, as could be understood from the documentation and examples.

### How to reproduce?

```
    @asset(op_tags={'dagster/concurrency_key': 'dummy', 'dagster/priority': '6'},
           group_name='dummy')
    def dummy_asset_2():
        sleep(10)
        return


    jobb = define_asset_job('dummy_asset_2_job',
                            [dummy_asset_2],
                            tags={'dagster/priority': '1'})


    @asset(op_tags={'dagster/concurrency_key': 'dummy', 'dagster/priority': '5'},
           group_name='dummy')
    def dummy_asset_1():
        sleep(10)
        return
```

```
```

### Dagster version

1.8.8

### Deployment type

None

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.
By submitting this issue, you agree to follow Dagster's [Code of Conduct](https://github.com/dagster-io/dagster/blob/master/.github/CODE_OF_CONDUCT.md).

**Comment by ion-elgreco** [comment]
Interesting, I have used this before on k8s and it worked well. It would move runs up in the queue when you had large backfills already running

**Comment by mmutso-boku** [comment]
Tested this again in 1.9.13 and it is still relevant.

How to reproduce:

dagster.yaml:
```yaml
concurrency:
  default_op_concurrency_limit: 1

run_coordinator:
  module: dagster.core.run_coordinator
  class: QueuedRunCoordinator
  config:
    max_concurrent_runs: 20
    block_op_concurrency_limited_runs:
      enabled: true
      op_concurrency_slot_buffer: 0

run_monitoring:
  enabled: true
  free_slots_after_run_end_seconds: 30

```

test_example.py
```python3
from time import sleep

from dagster import (
    Definitions,
    asset,
)


@asset(op_tags={'dagster/concurrency_key': 'dummy',
                'dagster/priority': '6'},
       group_name='dummy')
def dummy_asset_2():
    sleep(10)
    return


@asset(op_tags={'dagster/concurrency_key': 'dummy',
                'dagster/priority': '5'},
       group_name='dummy')
def dummy_asset_1():
    sleep(10)
    return


defs = Definitions(assets=[dummy_asset_1, dummy_asset_2])

```

Steps:
1) Start the Dagster DB and Dagster itself with `dagster dev -f test_example.py`
2) Spam the materialize button for dummy_asset_1 a number of times
3) Spam the materialize button for dummy_asset_2 a number of times

What is expected is that once the initial dummy_asset_1 run finishes, the next runs should be dummy_asset_2 runs since it has a higher priority defined, but in reality the runs are taken from the queue sequentially. It is as if the priority is ignored.

Am I misunderstanding something about the usage?

In version 1.8.8 documentation, there was this example:

![Image](https://github.com/user-attachments/assets/cdf1626d-522a-4298-8039-c068f8c28bde)
In the newer versions's documentation, I can no longer find this - does this mean that this is no longer intended to be used like this?

Basically would like to understand if this is a bug, or user error on my part.

**Comment by mmutso-boku** [comment]
@prha as the assignee on this issue, would you have any insight on the above?

**Comment by prha** [comment]
The asset/op priority does have an effect, but only for prioritizing concurrency slots for runs that are already in progress.  It does not affect which set of runs are queued vs in progress (if pool granularity is set to "run" or if you have other run-concurrency controls in place).
